### ISSUE TITLE
lint: unusedParam warns on external functions (no body)

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#264]
This is probably should be fixed in `wrapFuncDeclChecker`.
Proposed fix is to call `c.VisitFunc(decl)` and skip declaration if this call returns false.

Reproducer:
```
gocritic check-package -enable=unusedParam crypto/elliptic
```
Output:
```

src/crypto/elliptic/p256_amd64.go:101:52: unusedParam: parameter `sign` isn't used, consider to name it as `_`
src/crypto/elliptic/p256_amd64.go:101:58: unusedParam: parameter `sel` isn't used, consider to name it as `_`
src/crypto/elliptic/p256_amd64.go:101:63: unusedParam: parameter `zero` isn't used, consider to name it as `_`
src/crypto/elliptic/p256_amd64.go:101:28: unusedParam: parameter `res` isn't used, consider to name it as `_`
src/crypto/elliptic/p256_amd64.go:101:33: unusedParam: parameter `in1` isn't used, consider to name it as `_`
src/crypto/elliptic/p256_amd64.go:101:38: unusedParam: parameter `in2` isn't used, consider to name it as `_`
```
Code:
```go
// Point add with in2 being affine point
// If sign == 1 -> in2 = -in2
// If sel == 0 -> res = in1
// if zero == 0 -> res = in2
//go:noescape
func p256PointAddAffineAsm(res, in1, in2 []uint64, sign, sel, zero int)
```
This function has no body => no usages visible for `go/types`.
Most checkers skip func decls with empty bodies, this is why default `VisitFunc` implementation does exactly that.

Negative tests should be extended to cover this.
